Sources/NIOHTTP1/HTTPTypes.swift

```diff
@@ -174,17 +174,323 @@ struct HTTPHeader {
+    public static let timeout = HTTPHeaderName("timeout")
+    public static let title = HTTPHeaderName("title")
+    public static let topic = HTTPHeaderName("topic")
+    public static let trailer = HTTPHeaderName("trailer")
+    public static let transferEncoding = HTTPHeaderName("transfer-encoding")
+    public static let ttl = HTTPHeaderName("ttl")
+    public static let uaColor = HTTPHeaderName("ua-color")
+    public static let uaMedia = HTTPHeaderName("ua-media")
+    public static let uaPixels = HTTPHeaderName("ua-pixels")
+    public static let uaResolution = HTTPHeaderName("ua-resolution")
+    public static let uaWindowpixels = HTTPHeaderName("ua-windowpixels")
+    public static let urgency = HTTPHeaderName("urgency")
+    public static let uri = HTTPHeaderName("uri")
+    public static let upgrade = HTTPHeaderName("upgrade")
+    public static let userAgent = HTTPHeaderName("user-agent")
+    public static let variantVary = HTTPHeaderName("variant-vary")
+    public static let vary = HTTPHeaderName("vary")
+    public static let version = HTTPHeaderName("version")
+    public static let via = HTTPHeaderName("via")
+    public static let wwwAuthenticate = HTTPHeaderName("www-authenticate")
+    public static let wantDigest = HTTPHeaderName("want-digest")
+    public static let warning = HTTPHeaderName("warning")
+    public static let xContentTypeOptions = HTTPHeaderName("x-content-type-options")
+    public static let xDeviceAccept = HTTPHeaderName("x-device-accept")
+    public static let xDeviceAcceptCharset = HTTPHeaderName("x-device-accept-charset")
+    public static let xDeviceAcceptEncoding = HTTPHeaderName("x-device-accept-encoding")
+    public static let xDeviceAcceptLanguage = HTTPHeaderName("x-device-accept-language")
+    public static let xDeviceUserAgent = HTTPHeaderName("x-device-user-agent")
+    public static let xFrameOptions = HTTPHeaderName("x-frame-options")
+    public static let xRequestedWith = HTTPHeaderName("x-requested-with")
+    public static let xXssProtection = HTTPHeaderName("x-xss-protection")
+}
+
+/// Different types representing HTTP header names.
+private enum HTTPHeaderNameType {
+    /// Header name represented by a `String`.
+    case string(String)
+    /// Header name represented by a `HTTPHeaderName`.
+    case headerName(HTTPHeaderName)
+
+    /// Returns a lowercased version of the header name.
+    func lowercased() -> String {
+        switch self {
+        case .headerName(let name):
+            return name.lowerCaseName
+        case .string(let string):
+            return string.lowercased()
+        }
+    }
+}
+
+/// Extensions dealing with `HTTPHeaderNameType`s.
 private extension ByteBuffer {
-    func equalCaseInsensitiveASCII(view: String.UTF8View, at index: HTTPHeaderIndex) -> Bool {
-        guard view.count == index.length else {
+    /// Writes the `HTTPHeaderNameType` and also the separator (`: `). It returns the length of the name (without the seperator).
```

Comment:
Why does this return that strange number?

Reply:
@Lukasa just because thats what the is returned when `String` is used instead and so we not need to special case it.